pkg/splunk/controller/statefulset.go

```diff
@@ -230,3 +231,44 @@ func UpdateStatefulSetPods(c splcommon.ControllerClient, statefulSet *appsv1.Sta
 	scopedLog.Info("All pods are ready")
 	return splcommon.PhaseReady, nil
 }
+
+// SetStatefulSetOwnerRef sets owner references for statefulset
+func SetStatefulSetOwnerRef(client splcommon.ControllerClient, cr splcommon.MetaObject, namespacedName types.NamespacedName) error {
+
+	statefulset, err := GetstatefulSetByName(client, cr, namespacedName)
+	if err != nil {
+		return err
+	}
+
+	currentOwnerRef := statefulset.GetOwnerReferences()
+	// Check if owner ref exists
+	for i := 0; i < len(currentOwnerRef); i++ {
+		if reflect.DeepEqual(currentOwnerRef[i], splcommon.AsOwner(cr, false)) {
+			return nil
+		}
+	}
+
+	// Owner ref doesn't exist, update statefulset with owner references
+	statefulset.SetOwnerReferences(append(statefulset.GetOwnerReferences(), splcommon.AsOwner(cr, false)))
+
+	// Update owner reference if needed
+	err = splutil.UpdateResource(client, statefulset)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GetstatefulSetByName retrieves current statefulset
+func GetstatefulSetByName(c splcommon.ControllerClient, cr splcommon.MetaObject, namespacedName types.NamespacedName) (*appsv1.StatefulSet, error) {
```

Comment:
I don't think we need `cr` here as we are not using it in this function.

Reply:
my bad, I was playing around and forgot to change this. Let me fix this